scylla/src/transport/load_balancing/default.rs

```diff
@@ -297,29 +302,37 @@ or refrain from preferring datacenters (which may ban all other datacenters, if
                     &self.pick_predicate,
                     NodeLocationCriteria::DatacenterAndRack(dc, rack),
                 );
-                Either::Left(Self::round_robin_nodes(local_nodes, rack_predicate))
+                Either::Left(self.round_robin_nodes_with_shards(local_nodes, rack_predicate))
             } else {
-                Either::Right(std::iter::empty::<NodeRef<'a>>())
+                Either::Right(std::iter::empty::<(NodeRef<'a>, Shard)>())
             };
-        let robined_local_nodes = Self::round_robin_nodes(local_nodes, Self::is_alive);
+        let robined_local_nodes = self.round_robin_nodes_with_shards(local_nodes, Self::is_alive);
```

Comment:
Nit: this variable should be properly named `robinned`, or else it would be pronounced as *combined*.